### PR TITLE
Remove Ember.ContainerView

### PR DESCRIPTION
**Pull request by HeroicEric**
Continued work for https://github.com/emberjs/ember.js/pull/13188

Not sure if there's a way to add this to the other PR instead of creating a new one.

### CODE REVIEW

File: packages/ember-htmlbars/tests/integration/binding_integration_test.js [hunk 1]
```diff
@@ -123,57 +120,6 @@ QUnit.test('should cleanup bound properties on rerender', function() {
   equal(view.$().text(), 'wycats', 'rendered binding');
 });
 
-QUnit.test('should update bound values after view\'s parent is removed and then re-appended', function() {
```
Comment (HeroicEric) [inline packages/ember-htmlbars/tests/integration/binding_integration_test.js]: Should this really be removed or updated to work without using `Ember.ContainerView`?

File: packages/ember-views/tests/views/view/remove_test.js [hunk 2]
```diff
@@ -2,52 +2,16 @@ import { get } from 'ember-metal/property_get';
...
-
-
-// .......................................................
-// removeChild()
-//
+import { runAppend, runDestroy } from 'ember-runtime/tests/utils';
 
 var parentView, child;
 var originalViewKeyword;
-QUnit.module('View#removeChild', {
```
Comment (HeroicEric) [inline packages/ember-views/tests/views/view/remove_test.js]: Should this really be removed or updated to work without using `Ember.ContainerView`?

**Comment by HeroicEric** [comment]
@rwjblue not sure why the tests are failing on CI but they're all passing locally now. I'll look into why CI is failing.

**Comment by HeroicEric** [comment]
Hmm seems like they're failing with optional features enabled. Looking into it...

**Comment by HeroicEric** [comment]
@rwjblue Going to be stuck in this airport for another few hours. Anything else to work on here?

**Comment by rwjblue** [comment]
Nope, looks good to me!

**Pull request merged**